src/lib/reasoners/fun_sat.ml

```diff
@@ -1694,11 +1706,76 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
           "solved with backward!";
       raise e
 
+  let push env to_push =
+    if Options.get_tableaux_cdcl () then
+      Errors.run_error
+        (Errors.Unsupported_feature
+           "Incremental commands are not implemented in \
+            Tableaux(CDCL) solver ! \
+            Please use the Tableaux or CDLC SAT solvers instead"
+        );
+    let rec push_aux acc n =
+      if n <= 0 then acc
+      else
+        let b = E.fresh_name Ty.Tbool in
+        Stack.push b acc.incremental.stack_guard;
+
+        Stack.push (unit_facts acc) acc.unit_facts_cache;
+
+        let incremental =
+          { acc.incremental with
+            current_guard = Some b;
+            guards = SE.add b acc.incremental.guards;
+          } in
+        push_aux
+          {acc with incremental}
+          (n-1)
+    in
+    push_aux env to_push
+
+  let pop env to_pop =
+    if Options.get_tableaux_cdcl () then
+      Errors.run_error
+        (Errors.Unsupported_feature
+           "Incremental commands are not implemented in \
+            Tableaux(CDCL) solver ! \
+            Please use the Tableaux or CDLC SAT solvers instead"
+        );
+    let rec pop_aux acc n =
+      if n <= 0 then acc
+      else
+        let neg_b = Stack.pop acc.incremental.stack_guard in
+        let b =
+          if Stack.is_empty acc.incremental.stack_guard then
+            None
+          else Some (Stack.top acc.incremental.stack_guard)
+        in
```

Comment:
Don't understand why we have `let neg_b = ...` above.

Because, if `b` is of the form `Some c`, `c` is equal to `neg_b` (in both cases, we have `Stack.pop acc.incremental.stack_guard`)

Reply:
This was needed in an older version and it's not needed anymore